### PR TITLE
sqla 1.4 migration initialization

### PR DESCRIPTION
**Pull request by auvipy**
https://docs.sqlalchemy.org/en/14/changelog/migration_14.html#declarative-is-now-integrated-into-the-orm-with-new-features

### CODE REVIEW

File: examples/django/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 django>=2.2.1
-sqlalchemy>=1.0.14
+sqlalchemy==1.4
```
Comment (thedrow) [inline examples/django/requirements.txt:2]: We can relax this requirement can't we?
```suggestion
sqlalchemy>=1.4.0
```

File: celery/backends/database/session.py [hunk 2]
```diff
@@ -4,8 +4,7 @@
 from kombu.utils.compat import register_after_fork
 from sqlalchemy import create_engine
 from sqlalchemy.exc import DatabaseError
-from sqlalchemy.ext.declarative import declarative_base
-from sqlalchemy.orm import sessionmaker
+from sqlalchemy.orm import sessionmaker, declarative_base
```
Comment (thedrow) [inline celery/backends/database/session.py:7]: Why should we drop support for older versions?

**Review by thedrow — CHANGES_REQUESTED** [review]

File: celery/backends/database/session.py (hunk 2 above)
Comment (auvipy) [reply celery/backends/database/session.py:7]: declarative_base() - this is the same declarative base class that has been in use throughout the span of the declarative system, except it now references a registry object internally and is implemented by the registry.generate_base() method which can be invoked from a registry directly. The declarative_base() function creates this registry automatically so there is no impact on existing code. The sqlalchemy.ext.declarative.declarative_base name is still present, emitting a 2.0 deprecation warning when 2.0 deprecations mode is enabled.

Comment (auvipy) [reply celery/backends/database/session.py:7]: this is not a so big change from a sqla users perspective. as we are just updating the code path, nothing functional is changing with this and not in database level.

Comment (thedrow) [reply celery/backends/database/session.py:7]: I understand. But why should we drop support for older versions?

Comment (auvipy) [reply celery/backends/database/session.py:7]: sqla 1.4 is much slimmer in API & lay the groundwork for sqla 2.0 which will be python 3 only. if we are going to support multiple versions support, that won;t be hard as its just import and fallback to another version if import fail.

Comment (thedrow) [reply celery/backends/database/session.py:7]: I think we should support multiple versions until the next major version of Kombu.
It should be easy to do so.
Please add a TODO so we don't forget to remove the fallback once we drop support for sqlalchemy<1.4.

Comment (auvipy) [reply celery/backends/database/session.py:7]: ok

**Comment by auvipy** [comment]
https://github.com/celery/kombu/pull/1318

**Comment by thedrow** [comment]
I did it myself. I'm opening a PR right now.

**Pull request closed without merging**